test/compile-test.cc

```diff
@@ -171,3 +174,97 @@ TEST(CompileTest, TextAndArg) {
   EXPECT_EQ("42!", fmt::format(FMT_COMPILE("{}!"), 42));
 }
 #endif
+
+#if __cplusplus >= 202002L
+template <size_t max_string_length> struct constexpr_buffer_helper {
+  template <typename Func>
+  constexpr constexpr_buffer_helper& modify(Func func) {
+    func(buffer);
+    return *this;
+  }
+
+  template <typename T> constexpr bool operator==(const T& rhs) const noexcept {
+    return (std::string_view(rhs).compare(buffer.data()) == 0);
+  }
+
+  std::array<char, max_string_length> buffer{};
+};
```

Comment:
Can we turn this into a consteval formatting function that returns an array to simplify tests?

Something like:

```c++
template <size_t max_string_length, typename... Args>
std::array<char, max_string_length> test_format(const char* format_str, const Args&... args) {
  ...
}
```

Reply:
We could, but in that case, `FMT_COMPILE` should be outside of this `test_format` function, so every call would look like:
```cpp
constexpr auto result = test_format<5>(FMT_COMPILE("{}"), true);
```
Otherwise, I'm unable to create a compiled format from the `format_str` argument of `test_format`, it should have _baked_ string inside to be passed just as the type that can be created inside `compile.h` functions:
https://github.com/fmtlib/fmt/blob/b5dac0f0f8a112fba7b01165c792cd2aa8659c05/include/fmt/compile.h#L665-L668
But function argument is not enough because it's neither a string literal to use macro there nor a template parameter.

Reply:
But my attempts to find any solution (before I realized why it's impossible) ended up with some sort of replacement for `FMT_COMPILE` which is only available with C++20 of course (only trunk of Clang also): https://godbolt.org/z/of9j3s

And yes, even with this replacement of `FMT_COMPILE` the usage in our situation would look like:
```cpp
constexpr auto result = test_format<5>("{}"_cf, true);
```

Reply:
> FMT_COMPILE should be outside of this test_format function

Sure.

Reply:
> Can we turn this into a consteval formatting function that returns an array to simplify tests?

Not exactly `array` since a comparison operator is needed for GTest, but usage is simpler now

Reply:
Apparently, something like `_cf` was mentioned before https://github.com/fmtlib/fmt/issues/1874#issuecomment-716097084 and https://github.com/fmtlib/fmt/issues/613#issuecomment-491906397, and it works also on GCC 10. The `FMT_COMPILE` (not `FMT_STRING` since format can be checked with https://github.com/fmtlib/fmt/issues/2022) replacement will be my next PR probably if no one will do that 😏